app/components/common/TeamLeaderboardTable.component.js

```diff
@@ -1,53 +1,103 @@
+    );
+  };
+
+  return (
+    <div style={{ height: height }} className={"leaderboard"}>
+      <Table collapsing basic="very" className="main">
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell>#</Table.HeaderCell>
+            <Table.HeaderCell></Table.HeaderCell>
+            <Table.HeaderCell>
+              {isMainDashboard ? "Team Name" : "Member Name"}
+            </Table.HeaderCell>
+            <Table.HeaderCell>Distance</Table.HeaderCell>
+            {isMainDashboard && (
+              <Table.HeaderCell>Finish Date</Table.HeaderCell>
+            )}
+          </Table.Row>
+        </Table.Header>
+        {isLoading ? (
+          <Table.Body className={isMainDashboard ? "main-table loading" : ""}>
+            {_.times(8, () => (
+              <Table.Row>
+                <Table.Cell className={"loading"}></Table.Cell>
+              </Table.Row>
+            ))}
+          </Table.Body>
+        ) : (
+          <Table.Body className={isMainDashboard ? "main-table" : ""}>
+            {data.map((item, idx) => {
+              let dateCompletion = item.completionDate ? "finish" : "pending";
+              return (
+                <Table.Row
+                  key={item.name}
+                  className={dateCompletion}
+                  onClick={
+                    isMainDashboard ? () => handleClick(item.name) : undefined
+                  }
+                >
+                  <Table.Cell className={"main position"}>
+                    {item.position ? item.position : idx + 1}.
+                  </Table.Cell>
+                  <Table.Cell className={"main avatar"}>
+                    <Avatar
+                      name={item.name}
+                      color={isMainDashboard ? orange : blue}
+                      size={40}
+                    />
+                  </Table.Cell>
+                  <Table.Cell
+                    className={`main team-name ${!isMainDashboard && "team"}`}
+                  >
+                    {item.name}
+                  </Table.Cell>
+                  <Table.Cell
+                    className={`main distance ${
+                      !isMainDashboard && "team-view"
+                    }`}
+                  >
+                    {item.totalDistance}
+                    km
+                  </Table.Cell>
+                  {isMainDashboard && (
+                    <Table.Cell className={`main date ${dateCompletion}`}>
+                      {item.completionDate
+                        ? item.completionDate
+                        : "Still to finish"}
+                    </Table.Cell>
+                  )}
+                </Table.Row>
+              );
+            })}
+          </Table.Body>
+        )}
+      </Table>
+    </div>
+  );
+};
 
 TeamLeaderboardTable.propTypes = {
   height: PropTypes.number,
-  teams: PropTypes.array.isRequired,
+  data: PropTypes.array.isRequired,
+  isLoading: PropTypes.bool.isRequired,
```

Comment:
Might not need to set isLoading as isRequired. I don't think we have isLoading as a prop inside the team dashboard. Another option is to set isLoading=false for teams dashboard.

Reply:
We do have it on line 36 when I check for search results to load. If isLoading is true then it renders the animated loading table rows, if it is false it renders the table rows with the correct data.

Reply:
Based on that would be worth adding in TeamDashboard.component.js around line 67

Reply:
We only need the table rows to do the animation. If I added that to the TeamDashboard.component.js then it wouldn't display the table at all while searching.